src/main/scala/viper/gobra/frontend/Desugar.scala

```diff
@@ -433,10 +471,31 @@ object Desugar {
 
       val name = functionProxyD(decl, info)
       val fsrc = meta(decl)
+      val functionInfo = functionMemberOrLitD(decl, fsrc)
+
+      in.Function(name, functionInfo.args, functionInfo.results, functionInfo.pres, functionInfo.posts,
+        functionInfo.terminationMeasures, functionInfo.body)(fsrc)
+    }
 
+    private case class FunctionInfo(args: Vector[in.Parameter.In],
+                                    captured: Vector[(in.Expr, in.Parameter.In)],
+                                    results: Vector[in.Parameter.Out],
+                                    pres: Vector[in.Assertion],
+                                    posts: Vector[in.Assertion],
+                                    terminationMeasures: Vector[in.TerminationMeasure],
+                                    body: Option[in.MethodBody])
+
+    private def functionMemberOrLitD(decl: PFunctionOrClosureDecl, fsrc: Meta, outerCtx: FunctionContext = null): FunctionInfo = {
```

Comment:
I wouldn't use `null` here because it makes it tricky to understand all places where this `null` might propagate to. If you really need the possibility of not passing a `FunctionContext`, I would make this of type `Option[FunctionContext]`

Reply:
Good point, I opted for always requiring one (as a caller can always make a dummy FunctionContext anyway)